### PR TITLE
Fixed bug reporting in progress out of band, re-introducing InProgress workflow step

### PR DESCRIPTION
**Pull request by nihemstr**

### CODE REVIEW

File: src/adu_workflow/src/agent_workflow.c [hunk 1]
```diff
@@ -1327,6 +1336,36 @@ void ADUC_Workflow_MethodCall_Idle(ADUC_WorkflowData* workflowData)
...
+{
+    ADUC_WorkflowData* workflowData = methodCallData->WorkflowData;
+
+    ADUC_Result result = { .ResultCode = ADUC_Result_Success , .ExtendedResultCode = 0 };
+    Log_Info("Workflow step: ProcessDeployment");
+
+    //
+    // Shouldn't have to handle anything else here. WorkflowData already made?
+    //
+
```
Comment (jw-msft) [inline src/adu_workflow/src/agent_workflow.c:1355]: 2 newlines in a row. touch, stage, and run clang-format?

File: src/agent/src/main.c [hunk 2]
```diff
@@ -831,7 +831,7 @@ void ShutdownAgent()
  */
 void OnShutdownSignal(int sig)
 {
-    Log_Warn("Shutdown signal detected: %d", sig);
+    Log_Warn("Shutdown signal detected: %s", sig);
```
Comment (jw-msft) [inline src/agent/src/main.c]: the fix here seems to be regressing a previous fix. It should be %d to avoid undefined behavior (including segfault)

**Review by jw-msft — CHANGES_REQUESTED** [review]
I am blocking this PR because need to revert the change to line 834 on main.c which was previously fixing the undefined behavior / segfault.

File: src/agent/src/main.c (hunk 2 above)
Comment (jw-msft) [reply src/agent/src/main.c]: I've addressed this in https://github.com/Azure/iot-hub-device-update/commit/34e6d5cd2f25934b9370fc351c9532e663b741ca

**Review by jw-msft — APPROVED** [review]

**Review by jw-msft — APPROVED** [review]

**Review by Nox-MSFT — APPROVED** [review]

**Comment by ortogonal** [comment]
Just for information, I tested this PR and had some issues. See my comment here, https://github.com/Azure/iot-hub-device-update/issues/625#issuecomment-2435455883

**Comment by RichardHubertPtw** [comment]
Thanks for making some progress here! Please let us know if we can assist with tests or is there an alternative approach in the works? If so, please inform in the original issue ASAP. We are getting desperate :-)

**Pull request merged**